pallets/settlement/src/lib.rs

```diff
@@ -731,33 +736,35 @@ decl_module! {
 }
 
 impl<T: Config> Module<T> {
+    #[rustfmt::skip]
     fn lock_via_leg(leg: &Leg) -> DispatchResult {
-        match &leg.asset {
-            LegAsset::Fungible { ticker, amount } => {
-                T::Portfolio::lock_tokens(&leg.from, &ticker, *amount)
-            }
-            LegAsset::NonFungible(nfts) => with_transaction(|| {
```

Comment:
Do we still need to do the `with_transaction` part? I thought that all extrinsics were implicitly wrapped now with this?

Reply:
We do here. The caller of the function doesn't forward the error so we need the `with_transaction`. We could remove it here and place it in somewhere else, but I think we can leave that to a separate PR, since it's an important change on its own.